src/EnergyPlus/UnitarySystem.cc

```diff
@@ -0,0 +1,137 @@
+//     conditions and the following disclaimer.
+//
+// (2) Redistributions in binary form must reproduce the above copyright notice, this list of
+//     conditions and the following disclaimer in the documentation and/or other materials
+//     provided with the distribution.
+//
+// (3) Neither the name of the University of California, Lawrence Berkeley National Laboratory,
+//     the University of Illinois, U.S. Dept. of Energy nor the names of its contributors may be
+//     used to endorse or promote products derived from this software without specific prior
+//     written permission.
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <UnitarySystem.hh>
+#include <DataIPShortCuts.hh>
+#include <InputProcessing/InputProcessor.hh>
+#include <SimAirServingZones.hh>
+#include <UtilityRoutines.hh>
+
+namespace EnergyPlus {
+namespace UnitarySystems {
+
+    bool myOneTimeFlag(true);
+    bool getInputOnceFlag(true);
+    std::vector<UnitarySys> unitarySys;
+
+    void UnitarySys::simulate(std::string const & objectName, bool const firstHVACIteration)
+    {
+        UnitarySys::init(firstHVACIteration);
+    }
+
+    UnitarySys * UnitarySys::factory(int object_type_of_num, std::string const objectName)
+    {
+        if (getInputOnceFlag) {
+            UnitarySys mySys;
+            mySys.getInput();
+            getInputOnceFlag = false;
+        }
```

Comment:
I don't understand why I had to do this when the EIR WWHP did it this way:

    if (getInputsWWHP) {
         EIRWaterToWaterHeatPump::processInputForEIRWWHP();
         getInputsWWHP = false;
    }